control-plane-operator/controllers/hostedcontrolplane/v2/controlplaneoperator/podmonitor.go

```diff
@@ -12,6 +12,11 @@ import (
 
 func (cpo *ControlPlaneOperatorOptions) adaptPodMonitor(cpContext component.WorkloadContext, podMonitor *prometheusoperatorv1.PodMonitor) error {
 	podMonitor.Spec.NamespaceSelector = prometheusoperatorv1.NamespaceSelector{MatchNames: []string{cpContext.HCP.Namespace}}
+
+	if len(podMonitor.Spec.PodMetricsEndpoints) == 0 {
```

Comment:
Why are we adding all these blocks? `podMetricsEndpoints` should be set in any pod monitor manifest we add.  It isn't like there is a code path where these are conditionally set.